lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1520,6 +1558,43 @@ class NGPHistoryParser {
       }
     }
 
+    // Check if the last temp basal event is a start event. If we have a TEMP_BASAL_PROGRAMMED
+    // without a TEMP_BASAL_COMPLETE, we should synthesize the temp basal event with what we
+    // know, and it will be overwritten in the next upload that contains the corresponding
+    // TEMP_BASAL_COMPLETE event.
+    const lastTempBasalEvent =
+      this.eventsOfTypeReverse(
+        NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_PROGRAMMED,
+        NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_COMPLETE,
+      ).next().value;
+    if (lastTempBasalEvent &&
+      lastTempBasalEvent.eventType === NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_PROGRAMMED) {
+      const suppressedBasal = this.findSuppressedBasal(lastTempBasalEvent);
+      const basal = this.cfg.builder.makeTempBasal()
+        .with_rate(lastTempBasalEvent.rate)
+        .with_expectedDuration(lastTempBasalEvent.expectedDuration)
+        .with_duration(lastTempBasalEvent.expectedDuration)
+        .set('suppressed', {
```

Comment:
What happens if there is no suppressed basal? Does it just send up `null`?

Reply:
Yes, it would send up null. The only time I could see this happening is if the *only* basal event in the whole upload was a TEMP_BASAL_PROGRAMMED.